src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -214,10 +224,23 @@ public Task<CompletionItem> Handle(CompletionItem completionItem, CancellationTo
 
             if (markdown != null)
             {
+                MarkupKind markupKind;
+
+                var markupDocumentationFormat = LanguageServer.ClientSettings?.Capabilities?.TextDocument?.Completion.Value?.CompletionItem?.DocumentationFormat;
```

Comment:
By re-doing this logic here you're coupling the expectation that this and the description info factory are pulling from these capabilities. I think it'd be better to expand the `TagHelperDescriptionFactory` to provide `TagHelperDescription` types which have a markup kind associated:

```C#
var documentation = new StringOrMarkupContent(
	new MarkupContent()
	{
		Kind = tagHelperDescription.Kind,
		Value = tagHelperDescription.Documentation,
	});
completionItem.Documentation = documentation;
```

Also as a side note for the future. This class already has completion capability you could have pulled from.

Reply:
Good call, but is there some reason it shouldn't just be a `MarkupContent` object? That's what felt the most natural since both your suggested new class and `MarkupContent` have the same 2 properties and represent the same data.

Reply:
ah, good point. No reason at all go ahead and make it that 😄